sdks/python/apache_beam/runners/portability/portable_runner.py

```diff
@@ -129,11 +129,25 @@ def _create_environment(options):
               env=(config.get('env') or '')
           ).SerializeToString())
     elif environment_urn == common_urns.environments.EXTERNAL.urn:
+      def _looks_like_json(environment_config):
+        import re
+        return re.match(r'\{.+\}', environment_config)
```

Comment:
use `re.search()`.  also, I don't think this needs to be private, so remove the leading underscore.

maybe add a comment like:  "we don't use json.loads to test validity because we don't want to propagate json syntax errors downstream to the runner"

Reply:
Thanks! For the `re.search()` part, do we want it to be not searching from the start? I was thinking that if it is valid json string it should start and end with '{}', maybe `re.match(r'\{.+\}$')` or `re.search(r'^\{.+\}$')`?

Reply:
There's also whitespace to consider.  So if we're using `re.match` we have to do `re.match(r'\s*\{.*\}\s*$')`.  

We just need a simple heuristic that tells us that it looks json and not a url.  The most important thing is that we don't end up with a regex that incorrectly returns False for something that is *actually* json, so I suggested the `re.search` option because I think it's sufficient at detecting something that's *not a url* and thus probably json.  That said, I think the `re.match` regex above looks pretty safe.  Obviously, we're only talking about json objects (i.e. dict) and not arrays and other scalars.

Reply:
Just to reiterate, we want json syntax errors to occur here, at submission time, and not later on, so we want to answer "did the user attempt to pass a json map or a url?"  So I was even considering `return '{' in config or '}' in config`, since curly braces should not be in a url.  

I'll defer the final answer on this to the reviewer.

Reply:
Thanks Chad! Updated for now!